src/v2/style-guide/index.md

```diff
@@ -0,0 +1,1805 @@
+
+In committed code, prop definitions should always be as detailed as possible, specifying at least type(s).
+
+{% raw %}
+<details>
+<summary>
+  <h4>Detailed Explanation</h4>
+</summary>
+{% endraw %}
+
+Detailed [prop definitions](https://vuejs.org/v2/guide/components.html#Prop-Validation) have two advantages:
+
+- They document the API of the component, so that it's easy to see how the component is meant to be used.
+- In development, Vue will warn you if a component is ever provided incorrectly formatted props, helping you catch potential sources of error.
+
+{% raw %}</details>{% endraw %}
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+``` js
+// This is only OK when prototyping
+props: ['status']
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+``` js
+props: {
+  status: String
+}
+```
+
+``` js
+// Even better!
+props: {
+  status: {
+    type: String,
+    required: true,
+    validate: function (value) {
+      return [
+        'syncing',
+        'synced',
+        'version-conflict',
+        'error'
+      ].indexOf(value) !== -1
+    }
+  }
+}
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Keyed `v-for` <sup data-p="a">essential</sup>
+
+**Always use `key` with `v-for`.**
+
+It's _always_ required on components, in order to maintain internal component state down the subtree. Even for elements though, it's a good practice to maintain predictable behavior, such as [object constancy](https://bost.ocks.org/mike/constancy/) in animations.
+
+{% raw %}
+<details>
+<summary>
+  <h4>Detailed Explanation</h4>
+</summary>
+{% endraw %}
+
+Let's say you have a list of todos:
+
+``` js
+data: function () {
+  return {
+    todos: [
+      {
+        id: 1,
+        text: 'Learn to use v-for'
+      },
+      {
+        id: 2,
+        text: 'Learn to use key'
+      }
+    ]
+  }
+}
+```
+
+Then you sort them alphabetically. When updating the DOM, Vue will optimize rendering to perform the cheapest DOM mutations possible. That might mean deleting the first todo element, then adding it again at the end of the list.
+
+The problem is, there are cases where it's important not to delete elements that will remain in the DOM. For example, you may want to use `<transition-group>` to animate list sorting, or maintain focus if the rendered element is an `<input>`. In these cases, adding a unique key for each item (e.g. `:key="todo.id"`) will tell Vue how to behave more predictably.
+
+In our experience, it's better to _always_ add a unique key, so that you and your team simply never have to worry about these edge cases. Then in the rare, performance-critical scenarios where object constancy isn't necessary, you can make a conscious exception.
+
+{% raw %}</details>{% endraw %}
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+``` html
+<ul>
+  <li v-for="todo in todos">
+    {{ todo.text }}
+  </td>
```

Comment:
Why `</td>`?

Reply:
Because I forgot to change it when modifying the example. 😅  Good catch!